cpp/src/groupby/hash/groupby.cu

```diff
@@ -84,9 +94,114 @@ bool constexpr is_hash_aggregation(aggregation::Kind t)
   // return array_contains(hash_aggregations, t);
   return (t == aggregation::SUM) or (t == aggregation::MIN) or (t == aggregation::MAX) or
          (t == aggregation::COUNT_VALID) or (t == aggregation::COUNT_ALL) or
-         (t == aggregation::ARGMIN) or (t == aggregation::ARGMAX);
+         (t == aggregation::ARGMIN) or (t == aggregation::ARGMAX) or (t == aggregation::MEAN) or
+         (t == aggregation::SUM_OF_SQUARES) or (t == aggregation::STD) or
+         (t == aggregation::VARIANCE);
```

Comment:
This change is still not correct. You can't compute a `MEAN` with a hash aggregation. You can compute `SUM` and `COUNT` and then compute the `MEAN` from those intermediate results. Only aggregations that can be _directly_ performed with the hash aggregation should be listed here. Compound aggregations should not be.

Reply:
Actually that's similar to my original plan for calculating mean. In order to calculate STD using hash, we need to calculate the mean in sparse so that it can be picked by STD. So in a way if STD is a hash aggregation, MEAN also needs to be a hash aggregation.

Although, in this PR, it appears that STD doesn't depend on MEAN and calculates mean on its own.

Reply:
`detail::hash::can_use_hash_groupby` which uses this `constexpr is_hash_aggregation`, is used to take decision to take which hash route or sort route. So, that's the reason for adding these aggregation to the list here. 
https://github.com/rapidsai/cudf/blob/aba55dfb4cb205860f8fdfa14c323e7de1bea049/cpp/src/groupby/groupby.cu#L65-L69

If the comment is about, distinguishing compound hash aggregations from simple hash aggregations, another list can be added and maintained. Right now, that logic id done by simply checking by return values of `get_simple_aggregations()`.

Reply:
> which uses this constexpr is_hash_aggregation, is used to take decision to take which hash route or sort route.

Yes, and the compound aggregations should be preprocessed into their simple aggregations before `is_hash_aggregation` is used.